--- mupdf-1.3-source/Makerules.orig	2013-11-01 20:37:30.098722412 +0100
+++ mupdf-1.3-source/Makerules	2013-11-01 20:53:10.238682958 +0100
@@ -56,6 +56,8 @@
 endif
 
 SYS_CURL_DEPS = -lpthread -lrt
+SYS_CURL_CFLAGS = $(shell pkg-config --cflags libcurl)
+SYS_CURL_LIBS = $(shell pkg-config --libs libcurl) -lpthread
 
 SYS_X11_CFLAGS = $(shell pkg-config --cflags x11 xext)
 SYS_X11_LIBS = $(shell pkg-config --libs x11 xext)
--- mupdf-1.3-source/Makethird.orig	2013-11-01 20:54:35.492012714 +0100
+++ mupdf-1.3-source/Makethird	2013-11-01 20:54:07.405347226 +0100
@@ -444,5 +444,6 @@
 CURL_CFLAGS := -I$(CURL_DIR)/include
 CURL_LIBS := $(SYS_CURL_DEPS)
 else
-NOCURL := yes
+CURL_CFLAGS := $(SYS_CURL_CFLAGS)
+CURL_LIBS := $(SYS_CURL_LIBS)
 endif
