--- mupdf-1.8-source/source/pdf/js/pdf-jsimp-mu.c.orig	2015-11-10 17:19:51.000000000 +0100
+++ mupdf-1.8-source/source/pdf/js/pdf-jsimp-mu.c	2016-04-14 21:43:36.345879945 +0200
@@ -305,7 +305,7 @@
 void pdf_jsimp_execute(pdf_jsimp *imp, char *code)
 {
 	js_State *J = imp->J;
-	js_dostring(J, code, 0);
+	js_dostring(J, code);
 }
 
 void pdf_jsimp_execute_count(pdf_jsimp *imp, char *code, int count)
